python/cudaq/runtime/sample.py

```diff
@@ -66,8 +69,17 @@ def sample(kernel,
   or a list of such results in the case of `sample` function broadcasting."""
 
     has_conditionals_on_measure_result = False
-    if hasattr(kernel, 'metadata') and kernel.metadata.get(
-            'conditionalOnMeasure', False):
+
+    if isinstance(kernel, PyKernelDecorator):
+        kernel.compile()
+        if kernel.module is not None:
+            for operation in kernel.module.body.operations:
+                if not hasattr(operation, 'name'):
+                    continue
+                if nvqppPrefix + kernel.name == operation.name.value and 'qubitMeasurementFeedback' in operation.attributes:
+                    has_conditionals_on_measure_result = True
+                    break
```

Comment:
nit: I think we can always break here regardless of the `and` condition, e.g.,
```suggestion
                if nvqppPrefix + kernel.name == operation.name.value:
                    has_conditionals_on_measure_result = 'qubitMeasurementFeedback' in operation.attributes
                    break
```